firmware/README.md

```diff
@@ -0,0 +1,61 @@
+# Introduction
+
+Open-TL866 operates using *modes*. Each mode supports one or
+more target chips. A mode is used by uploading the .hex file
+for that mode to the TL866 (see
+[parent README](../README.md)), and then running the
+corresponding Python command line tool from the `py/otl866`
+directory.
+
+# Supporting a new mode
+
+To support a new mode:
+
+1.  Create a directory in `firmware/modes` for your new mode.
+
+1.  Create a `main.c` file in your mode's directory. It is
+    best to start by copying it from one of the other modes.
+
+1.  Create a `<mode>.h` and `<mode>.c` file in `firmware`
+    for your new mode.
+
+1.  Define low-level primitives in your `<mode>.*` files, using
+    the basic calls in `io.h`
+    (see [Firmware API](#Firmware-API)).
+
+1.  Define text commands in your `main.c` file to call the
+    low-level primitives.
+
+1.  Write a Python script `py/otl866/<mode>.py` to connect to
+    your firmware and execute your commands. It is best to
+    start by copying your script from one of the other modes.
+
+1.  Test, test, test.
+
+# Firmware API
+
+For ZIF pin voltage supply capabilities, see
+[pin_mapping.md](pin_mapping.md).
+
+The most useful API is specified in `firmware/io.h`. See that
+file for detailed documentation. Briefly:
```

Comment:
Concerned this is redundant with the .c/.h file. Also this will likely get out of date as people update the .c/.h and not this

Reply:
Maybe it's better to reorganize io.h so that the useful public functions are at the top, and then just point to that file. With the doxy-style comments, that's pretty much your API documentation.